### PR TITLE
[FlangClassic]: Fix a couple of warnings

### PR DESCRIPTION
**Pull request by nextsilicon-itay-bookstein**
This PR fixes a couple of warnings that pop up in flang patches when compiling with e.g. clang-12.

### CODE REVIEW

**Comment by nextsilicon-itay-bookstein** [comment]
ping :)

**Comment by nextsilicon-itay-bookstein** [comment]
Ping :)

**Comment by kiranchandramohan** [comment]
Apologies, will have a look this week.

**Review by pawosm-arm — APPROVED** [review]

**Comment by pawosm-arm** [comment]
Remember to cherry pick this to LLVM13!

**Comment by nextsilicon-itay-bookstein** [comment]
ping :) (should I reopen this on release_13x or will you just cherrypick it?)

**Comment by pawosm-arm** [comment]
@nextsilicon-itay-bookstein I think you'd have to negotiate this with @abrahamtovarmob

**Comment by abrahamtovarmob** [comment]
@nextsilicon-itay-bookstein I will cherry pick'ed into release_13x, no worries.

**Comment by nextsilicon-itay-bookstein** [comment]
Ping :)

File: llvm/lib/CodeGen/AsmPrinter/DwarfCompileUnit.h [hunk 1]
```diff
@@ -151,7 +151,6 @@ class DwarfCompileUnit final : public DwarfUnit {
       DenseMap<DIGlobalVariable *, SmallVector<GlobalExpr, 1>> *p = nullptr) {
```
Comment (bryanpkc) [inline llvm/lib/CodeGen/AsmPrinter/DwarfCompileUnit.h]: Please also delete `globalVarMap` and `setGlobalVarMap`. They serve no purpose any more.

Comment (nextsilicon-itay-bookstein) [reply llvm/lib/CodeGen/AsmPrinter/DwarfCompileUnit.h]: Done.

**Review by bryanpkc — APPROVED** [review]
LGTM.

**Comment by bryanpkc** [comment]
@nextsilicon-itay-bookstein  Do you have any idea why the MacOS tests failed?

**Comment by nextsilicon-itay-bookstein** [comment]
Looks like an unrelated issue that was fixed in LLVM 13 by https://reviews.llvm.org/D104332
From the summary:
> Since Xcode 13, ld64 requires linking libSystem for all the executable.
> Fix the tests that needs to run ld64 by linking libSystem from sysroot.

Perhaps it was caused by an upgrade to the CI's build tools? Cherry-picking it would probably fix the issue; how would you like to proceed? (should I cherry-pick it into this PR?)

**Comment by bryanpkc** [comment]
The macOS job uses the AppleClang 13.0.0.13000029 toolchain, which provides the `ld64` command used by `llvm-lit`. As you said, cherry-picking that fix into release_12x will probably work. Please go ahead and add that patch in this PR.

**Comment by nextsilicon-itay-bookstein** [comment]
Done.

**Comment by nextsilicon-itay-bookstein** [comment]
The cherry-pick seems to have solved the issue.

**Comment by bryanpkc** [comment]
@shivaramaarao  Can you review and approve this please?

**Review by shivaramaarao — APPROVED** [review]

**Comment by bryanpkc** [comment]
Thanks @nextsilicon-itay-bookstein. Please submit a PR with the same changes for the `release_13x` branch.

**Pull request merged**